server/utils/csvDownloaderUtil.js

```diff
@@ -0,0 +1,23 @@
+import { stringify } from 'csv-stringify'
+
+export default async function csvDownloader(res, { tableColumns, tableRows }) {
+  const tableDataToCsvString = stringify(tableRows, {
+    header: true,
+    columns: tableColumns,
+  })
+
+  tableDataToCsvString.on('readable', function () {
+    let row
+    while ((row = tableDataToCsvString.read()) !== null) {
+      res.write(row)
+    }
+  })
+
+  tableDataToCsvString.on('error', function (err) {
+    return new Error(err)
```

Comment:
should this throw the error instead of returning it?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/287/commits/53c75c19215f81ea1f646fb5e98102019a21521f